test/deployframework/local.go

```diff
@@ -0,0 +1,206 @@
+	MeteringOperatorPID      int
+	ReportingOperatorPID     int
+	ReportingOperatorTLSPath string
+	Logger                   logrus.FieldLogger
+}
+
+// RunMeteringOperatorLocal is a method that runs the metering-operator locally
+func (lc *LocalCtx) RunMeteringOperatorLocal() error {
+	dockerCmd := exec.Command(
+		"docker", "run",
+		"--rm",
+		"-u", "0:0",
+		"--name", meteringOperatorContainerName,
+		"-v", lc.KubeConfigPath+":/kubeconfig",
+		"-v", ansibleRunnerPath,
+		"-e", "KUBECONFIG="+destKubeConfigPath,
+		"-e", "OPERATOR_NAME="+meteringOperatorName,
+		"-e", "POD_NAME="+meteringOperatorName,
+		"-e", "WATCH_NAMESPACE="+lc.Namespace,
+		"-e", "ENABLE_DEBUG="+strconv.FormatBool(enableDebug),
+		"-e", "DISABLE_OCP_FEATURES="+strconv.FormatBool(disableOCPFeatures),
+		lc.MeteringOperatorImage,
+	)
+
+	lc.Logger.WithFields(logrus.Fields{
+		"command": dockerCmd.Args,
+	}).Infof("Running the metering-operator docker container")
+
+	logFile, err := ioutil.TempFile(lc.BasePath, meteringOperatorLogName)
+	if err != nil {
+		return fmt.Errorf("failed to create the metering-operator container output log file: %v", err)
+	}
+
+	lc.Logger.Infof("Storing the metering-operator container logs to the '%s' path", logFile.Name())
+
+	dockerCmd.Stdout = logFile
+	dockerCmd.Stderr = logFile
+
+	err = dockerCmd.Start()
+	if err != nil {
+		return fmt.Errorf("failed to run the metering-operator container: %v", err)
+	}
+
+	lc.MeteringOperatorPID = dockerCmd.Process.Pid
+
+	return nil
+}
+
+// RunReportingOperatorLocal is a method that runs the reporting-operator locally
+func (lc *LocalCtx) RunReportingOperatorLocal(apiListenPort, metricsListenPort, pprofListenPort int, token string) error {
+	var err error
+
+	// TODO: should this still be a tmpdir in the case where tests are interrupted early?
+	lc.ReportingOperatorTLSPath, err = ioutil.TempDir(lc.BasePath, "reporting-operator-tls")
+	if err != nil {
+		return fmt.Errorf("failed to create the reporting-operator TLS directory: %v", err)
+	}
+
+	envVarArr := []string{
+		"METERING_NAMESPACE=" + lc.Namespace,
+		"METERING_USE_SERVICE_ACCOUNT_AS_PROM_TOKEN=false",
+		"REPORTING_OPERATOR_PROMETHEUS_BEARER_TOKEN=" + token,
+		"REPORTING_OPERATOR_API_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, apiListenPort),
+		"REPORTING_OPERATOR_METRICS_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, metricsListenPort),
+		"REPORTING_OPERATOR_PPROF_LISTEN=" + fmt.Sprintf("%s:%d", localAddr, pprofListenPort),
+		"REPORTING_OPERATOR_PROMETHEUS_DATASOURCE_MAX_IMPORT_BACKFILL_DURATION=15m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=30s",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_CHUNK_SIZE=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_INTERVAL=5m",
+		"REPORTING_OPERATOR_PROMETHEUS_METRICS_IMPORTER_STEP_SIZE=60s",
+		"REPORTING_OPERATOR_TLS_DIR=" + lc.ReportingOperatorTLSPath,
+	}
+
+	// TODO: add this to LocalCtx?
+	scriptPath, err := filepath.Abs("../../hack/run-reporting-operator-local.sh")
```

Comment:
yeah probably ideal if we have a flag/variable (defaulting to PWD) that's the path to the repo. I think deployer.Config has similar stuff for manifests. Then we can just use `path.Join(repoDir, "hack/run-reporting-operator-local.sh")`. What do you think?

Reply:
I think it may make more sense to provide a path to the `hack/` directory instead, as we also need the path to the `hack/run-test-cleanup.sh` script that logs the resource and container logs. 

That way we can remove the cleanup script flag, and only manage the path to the `hack` script directory directly (via `filepath.Join(...)` and const variables).

Reply:
I was thinking the path to the repo (which could default to `PWD`) since we already need the manifest directory, which is in the repo and we could potentially just use one var in tests for paths, and it could just be the path to the repo. What do you think?

Reply:
After playing around with this a bit, I think it would make the most sense to default to `../../` which assumes you're running the `go test ...` command directly (as hack/e2e.sh sets this to `$ROOT_DIR`).

If you elect to run go test directly, then you would need to provide the absolute path to the operator-metering directory.